ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -92,113 +101,77 @@ DdsPipe::~DdsPipe()
 utils::ReturnCode DdsPipe::reload_allowed_topics(
         const std::shared_ptr<AllowedTopicList>& allowed_topics)
 {
-    std::lock_guard<std::recursive_mutex> lock(mutex_);
+    std::lock_guard<std::mutex> lock(mutex_);
+
+    logDebug(DDSROUTER, "Reloading DDS Pipe configuration...");
 
-    if (enabled_.load())
+    // Check if it should change or is the same configuration
+    if (*allowed_topics == *allowed_topics_)
     {
-        logDebug(DDSROUTER, "Reloading DDS Pipe configuration...");
+        logDebug(DDSROUTER, "Same configuration, do nothing in reload.");
+        return utils::ReturnCode::RETCODE_NO_DATA;
+    }
 
-        // Check if it should change or is the same configuration
-        if (*allowed_topics == *allowed_topics_)
-        {
-            logDebug(DDSROUTER, "Same configuration, do nothing in reload.");
-            return utils::ReturnCode::RETCODE_NO_DATA;
-        }
+    // Set new Allowed list
+    allowed_topics_ = allowed_topics;
 
-        // Set new Allowed list
-        allowed_topics_ = allowed_topics;
+    logDebug(DDSROUTER, "New DDS Pipe allowed topics configuration: " << allowed_topics_);
 
-        logDebug(DDSROUTER, "New DDS Pipe allowed topics configuration: " << allowed_topics_);
+    if (!enabled_)
+    {
+        return utils::ReturnCode::RETCODE_OK;
```

Comment:
NIT: it is more accurate to return RETCODE_NOT_ENABLED

Reply:
I do not agree. This method is independent of the object state, so calling it being disabled should have a correct behaviour, as calling it being enabled